### ISSUE TITLE
poetry fails to find packages in private repositories - depends on venv

### ISSUE THREAD

**Issue by sandroden** [python-poetry/poetry#2178]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: ubuntu 19.10
- **Poetry version**: 1.0.5

## Issue
It seems that poetry fails to find the package from a private repository (both with nginx + autoindex and pypiserver) if virtualenv is created by poetry or by python -m venv. It finds it if the virtualenv is build with virtualenvwrapper.

I was trying to add package thx-nonamespace from repository `pypi.thux.dev` in a package created with `poetry new bianco` after adding support for a personal repository:

```
[tool.poetry]
name = "bianco"
version = "0.1.0"

[[tool.poetry.source]]
name = "dev"
url = "https://pypi.thux.dev/simple/"

```

and the error message is:

```
$ poetry add thx-nonamespace
Using version ^0.1.0 for thx-nonamespace

Updating dependencies
Resolving dependencies... (0.6s)


Package operations: 1 install, 0 updates, 0 removals

  - Installing thx-nonamespace (0.1.0)

[EnvCommandError]
Command ['/tmp/bianco/lotest/bin/pip', 'install', '--no-deps', '--index-url', 'https://thux:TFCxdr@pypi.thux.dev/simple', '--extra-index-url', 'https://pypi.org/simple/', 'thx-nonamespace==0.1.0'] errored with the following return code 1, and output: 
Looking in indexes: https://pypi.thux.dev/simple, https://pypi.thux.dev/simple/, https://pypi.org/simple/
Looking in links: https://pypi.thux.it
Collecting thx-nonamespace==0.1.0
Could not install packages due to an EnvironmentError: 404 Client Error: Not Found for url: https://pypi.org/simple/thx-nonamespace/
```

poetry start correctly detecting the existing release of pkg `thx-nonamespace` but then tries to downloading it from pypi.org and clearly fails to find it. This happens both with poestry's navive virtualenv and with virtualenv created with `python -m venv`.

**Everything works correctly if I use a virtualenv build with virtualenvwrapper**:

```
sandro@bluffx:/tmp/bianco$ mkvirtualenv junk-bianco
Running virtualenv with interpreter /usr/bin/python3.7
Using base prefix '/usr'
/usr/lib/python3/dist-packages/virtualenv.py:1090: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
New python executable in /home/sandro/.virtualenvs/junk-bianco/bin/python3.7
Also creating executable in /home/sandro/.virtualenvs/junk-bianco/bin/python
Please make sure you remove any previous custom paths from your /home/sandro/.pydistutils.cfg file.
Installing setuptools, pkg_resources, pip, wheel...done.
virtualenvwrapper.user_scripts creating /home/sandro/.virtualenvs/junk-bianco/bin/predeactivate
virtualenvwrapper.user_scripts creating /home/sandro/.virtualenvs/junk-bianco/bin/postdeactivate
virtualenvwrapper.user_scripts creating /home/sandro/.virtualenvs/junk-bianco/bin/preactivate
virtualenvwrapper.user_scripts creating /home/sandro/.virtualenvs/junk-bianco/bin/postactivate
virtualenvwrapper.user_scripts creating /home/sandro/.virtualenvs/junk-bianco/bin/get_env_details
(junk-bianco) sandro@bluffx:/tmp/bianco$ poetry add thx-nonamespace
Using version ^0.2.0 for thx-nonamespace

Updating dependencies
Resolving dependencies... (0.7s)


Package operations: 12 installs, 0 updates, 0 removals

  - Installing zipp (3.1.0)
  - Installing importlib-metadata (1.5.0)
  - Installing pyparsing (2.4.6)
  - Installing six (1.14.0)
  - Installing attrs (19.3.0)
  - Installing more-itertools (8.2.0)
  - Installing packaging (20.3)
  - Installing pluggy (0.13.1)
  - Installing py (1.8.1)
  - Installing wcwidth (0.1.8)
  - Installing pytest (5.3.5)
  - Installing thx-nonamespace (0.2.0)
```

**Comment by mrosales** [comment]
I also encountered this issue and the problem seems to be caused by the fact that its creating the virtualenv with the system version of python/pip. On Ubuntu LTS 18.04, the provided pip version via `python3-pip` is 9.0.1 😱. One of the fun thing that comes along with older pip versions is that it prints your passwords in plaintext... (@sandroden I'd recommend changing your password)

I can unblock myself by by always running:

```
poetry run pip install --upgrade pip setuptools wheel
poetry install
```

But I can't seem to specify a way to require a more recent pip. I tried using the following to ensure a new version of pip was installed but it didn't seem to have any effect.

```
[build-system]
requires = ["poetry>=1.0.0", "pip>=20.0.0"]
build-backend = "poetry.masonry.api"
```

**Comment by jonapich** [comment]
I've found that upgrading system pip using pip is a bad idea in the past; not sure if that changed nowdays.

Anyway, my `pip` (py2) and `pip3` both reports `20.0.1` (and so does `python -m pip` and its py3 equivalent) and i'm running into this issue as well.

**Comment by billkouts** [comment]
Hello all.
Pretty similar case with me as well

What I can do for a work around is run
```shell
poetry run pip install my_library_in_my_private_repo --index-url "https://private_hosting_service_of_wheels.org"
```
for every one of my dependencies.
Not ideal.

> I've found that upgrading system pip using pip is a bad idea in the past; not sure if that changed nowdays.

I agree with @jonapich here. Anyone going down that path should be fully aware that there may be side effects in any other part of the system.

(Main reason of me commenting is to +1 this issue.)

**Comment by sandroden** [comment]
Things are getting worse. Now even the case reported above where I could add from a private repository are failing. I'm not aware of any change in my setup. This bug makes it totally unusable for my projects. Updating to latest pip (I was already at 20.1) didn't turne useful.

**Comment by sandroden** [comment]
Good news: I have now a situation that works. The last step was to use a private repo using `pypiserver` rather than the private repository I used for years that just uses `nginx + autoenv`. I think this is acceptable but needs to be reported in the documentation.

When I tested the different setup I didn't realize that `poetry.lock` holds the `pyproject.toml` configuration and particularly the sources. Poetry adds a warning that `poetry.lock` needs an update, but I may have not put enough attention on it so that I thought it worked with both repositories but I was just working with the same one (pypiserver).

So, the working situation for me passes through these steps:

1. the private repo pypiserver (rather than nginx + autoenv)
2. `pip run pip inistall --upgrade pip`
   (@mrosales suggests also `setuptools` and `wheel`)
3. (clearly) add the repo in `pyproject.toml`: `[[tool.poetry.source]]`
